File: 1894-merge-strings-alternately/1894-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string res = "";
        int w1 = word1.size(), w2 = word2.size();
        int k = w1 + w2 - 1;
        int a = 0, b = 0;

        for(int i = 0; i <= k; i++) {
            if(i%2 == 0) {
                if(a < w1) {
                    res += word1[a];
                    a++;
                }
            } else {
                if(b < w2) {
                    res += word2[b];
                    b++;
                }
            }
        }
        
        while(a < w1) {
            res += word1[a];
            a++;
        }

        while(b < w2) {
            res += word2[b];
            b++;
        }

        return res;
    }
};

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res = ""
        a, b = 0, 0
        l1, l2 = len(word1), len(word2)

        while a < l1 and b < l2:
            if len(res)%2 == 0:
                res += word1[a]
                a += 1
            else:
                res += word2[b]
                b += 1
        
        while a < l1:
            res += word1[a]
            a += 1
        while b < l2:
            res += word2[b]
            b += 1
        
        return res

        return res